### ISSUE TITLE
MdTextViewLineCollection does not format all ITextViewLines in the view

### ISSUE THREAD

**Issue by gundermanc** [mono/monodevelop#3999]
Calling ITextViewLinesCollection.GetTextViewLineContainingBufferPosition() on anything but the first line in the view returns null, and the ITextViewLinesCollection.FormattedSpan does not encompass the span of text currently visible in the view.

In VS proper, the view's TextViewLinesCollection backs the actual rendering of the view, however in MD, it seems this dependency does not yet exist, meaning that nothing forces the generation of these ITextViewLines in the collection.

This breaks any extension or code that was built with VS proper in mind because many extensions that render adornments in the view only do so for TextViewLines that already exist in the TextViewLinesCollection. All of these extensions will fail to draw on anything but the first line in MD.

One such example that is broken is VS's async quick info support in MD's SourceEditor2 AddIn. Quick Info fails to show up for anything except for the first line of the TextView.

See MouseTrackingWpfQuickInfoPresenter.GetViewSpanRect(), which is the method that is currently broken.

Some of the affected usages are shown below:
```
                var startLine = view.TextViewLines.GetTextViewLineContainingBufferPosition(visualSpan.Start);

                if (startLine != null)

                {

                    var startPointBounds = startLine.GetExtendedCharacterBounds(visualSpan.Start);

                    if ((startPointBounds.Left < right) &&

                        (startPointBounds.Left >= view.ViewportLeft) &&

                        (startPointBounds.Left < view.ViewportRight))

                    {

                        left = startPointBounds.Left;

                    }

                }
```

> VS bug #576772

**Comment by gundermanc** [comment]
Actually, my original diagnosis might have been incorrect. MD is indeed generating a full TextViewLinesCollection, but there are a couple of distinct issues going on:

- TextViewLinesCollection is being regenerated each time it is requested instead of persisting and being updated on layout. This means that every time we get TextViewLines a different set of lines is potentially returned and there is work being done to create this collection. Many extensions call this getter repeatedly, so there is benefit to caching it.
- The following line in MdTextViewLineCollection.cs evaluates to '1' when called via the async quick info code path:

``` 
double lastY = textEditor.VAdjustment.Value + textEditor.Allocation.Height;
```

This results in the initialization loop terminating prematurely and there being only one line in the TextViewLinesCollection when we ask for it.

**Comment by gundermanc** [comment]
For help in debugging this issue, I've put together a repro project for async quick info: https://github.com/gundermanc/vs-xplat-extensibility-samples

Can't get it to build in the IDE, but it builds and deploys on the command line via the instructions in the README.

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.81 (master)

Pull Request #4056 merged by: Mike Krüger
Author: mono
Commit: 213baaf60346255bce4c9266e55391f3cbd4461a (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #4056 merged by: Mike Kr??ger
Author: mono
Commit: 213baaf60346255bce4c9266e55391f3cbd4461a (xamarin/monodevelop)
